File: integration-test/java/jee/src/main/java/com/qualityminds/lazyval/integration/domain/Order.java
package com.qualityminds.lazyval.integration.domain;

import com.qualityminds.lazyval.integration.shared.Isbn;
import com.qualityminds.lazyval.integration.shared.Quantity;

import java.util.UUID;

public record Order(UUID id, Isbn isbn, Quantity quantity, EMail email, OrderDate orderDate) {

    public static Order create(Isbn isbn, Quantity quantity, EMail email){
        return new Order(UUID.randomUUID(), isbn, quantity, email, OrderDate.now());
    }
}

File: integration-test/kotlin/quarkus-app/src/main/kotlin/com/qualityminds/lazyval/integration/domain/Order.kt
package com.qualityminds.lazyval.integration.domain

import com.qualityminds.lazyval.integration.shared.Isbn
import com.qualityminds.lazyval.integration.shared.Quantity
import java.util.UUID

data class Order(
    val id: UUID,
    val isbn: Isbn,
    val quantity: Quantity,
    val email: EMail,
    val orderDate: OrderDate
) {
    companion object {
        @JvmStatic
        fun create(isbn: Isbn, quantity: Quantity, email: EMail): Order =
            Order(UUID.randomUUID(), isbn, quantity, email, OrderDate.now())
    }
}
